common/helpers/field.js

```diff
@@ -1,12 +1,23 @@
 const { cloneDeep, fromPairs, get, set } = require('lodash')
 
 const componentService = require('../services/component')
-
-function mapReferenceDataToOption({ id, title, key, conditional }) {
 const i18n = require('../../config/i18n')
+
+function mapReferenceDataToOption({
+  id,
+  hint,
+  title,
+  key,
+  conditional,
+  label,
+}) {
   const option = {
     value: id,
-    text: title,
+    text: label || title,
```

Comment:
I wonder if we should leave this logic as it was as the majority of cases that are mapping reference data to an option will still expect to use title.

And then the logic for finding a "label" translation key should be kept to preceding map functions like the one in the previous commit. And that's where we handle the fallback.